api/apps/api/src/modules/cost-surface/delete-cost-surface/delete-cost-surface.handler.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { FakeLogger } from '@marxan-api/utils/__mocks__/fake-logger';
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { Injectable } from '@nestjs/common';
+import { CqrsModule, EventBus, IEvent } from '@nestjs/cqrs';
+import { Test } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { isLeft, isRight } from 'fp-ts/lib/Either';
+import { v4 } from 'uuid';
+import { DeleteCostSurfaceHandler } from './delete-cost-surface.handler';
+import { CostSurfaceDeleted } from '@marxan-api/modules/cost-surface/events/cost-surface-deleted.event';
+import {
+  DeleteCostSurfaceCommand,
+  deleteCostSurfaceFailed,
+  DeleteCostSurfaceResponse,
+} from '@marxan-api/modules/cost-surface/delete-cost-surface/delete-cost-surface.command';
+import { CostSurface } from '@marxan-api/modules/cost-surface/cost-surface.api.entity';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+
+it('deletes a Cost Surface and emits a CostSurfaceDeleted event', async () => {
+  const projectId = fixtures.GivenProjectExits();
+
+  const result = await fixtures.WhenACostSurfaceIsDeleted(projectId);
+
+  fixtures.ThenTrueIsReturned(result);
+  await fixtures.ThenCostSurfaceIsDeleted(projectId);
+  fixtures.ThenACostSurfaceDeletedEventIsEmitted(projectId);
+});
+
+it('fails to delete a project ', async () => {
+  const projectId = fixtures.GivenProjectExits();
+  fixtures.GivenDeleteOperationFails();
+
+  const result = await fixtures.WhenACostSurfaceIsDeleted(projectId);
+
+  fixtures.ThenDeleteCostSurfaceFailedIsReturned(result);
+  await fixtures.ThenCostSurfaceIsNotDeleted(projectId);
+  fixtures.ThenNoEventIsEmitted();
+});
```

Comment:
very minor (feel free to keep as is):
- I guess the test description is a leftover from copy/paste - it should likely be like `won't delete cost data when deleting the relevant cost surface fails` maybe?
- `GivenDeleteOperationFails()` - I had to follow the code to understand what this does - maybe `GivenCostSurfaceDeletionOperationWillFail()` or something like this?
- `ThenNoEventIsEmitted()` it is technically exactly what the function is checking, though if we want to make what is happening clearer, maybe we should call this `ThenNoCostSurfaceDeletedEventIsEmitted()`

Reply:
First one seems like the description might have been incomplete from the get go, and I had forgotten about it.
The other two, I thought it was implied enough from the context, but I see what you mean and I  do agree.
These changes seem minor enough that I could sneak them in another PR once this one is merged, don't they?

Reply:
yes, let's just get this merged, thanks!